pkg/types/chain_writer.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/google/uuid"
+)
+
+type ChainWriter interface {
+	// SubmiSignedTransaction packs and broadcasts a transaction to the underlying chain.
+	//
+	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
+	SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) (int64, error)
+
+	// StatusForUUID returns the current status of a transaction in the underlying chain's TXM.
+	StatusForUUID(ctx context.Context, transactionID uuid.UUID) (TransactionStatus, error)
+
+	// GetFeeComponents retrieves the associated gas costs for executing a transaction.
+	GetFeeComponents() (ChainFeeComponents, error)
+}
+
+// TxMeta contains metadata fields for a transaction.
+//
+// Eventually this will replace, or be replaced by (via a move), the `TxMeta` in core:
+// https://github.com/smartcontractkit/chainlink/blob/dfc399da715f16af1fcf6441ea5fc47b71800fa1/common/txmgr/types/tx.go#L121
+type TxMeta map[string]string
```

Comment:
Declaring this as a named type will require some explicit casts. What about using an alias?
```suggestion
type TxMeta = map[string]string
```
I don't grok the comment about future work though. Are you planning to extend this type with methods?

Reply:
An alias makes sense. Happy to make the change.

Reply:
There is a similar type currently being used in core (see the link). This will most likely be used in a similar manner, but extracting the struct from `core` is out of scope for this PR.